src/GeositeFramework/plugins/layer_selector_v2/WmsService.js

```diff
@@ -0,0 +1,139 @@
+define([
+        "dojo/_base/declare",
+        "dojo/Deferred",
+        "underscore",
+        "./lib/xmlToJson",
+        "framework/util/ajax",
+        "./util",
+        "./LayerNode",
+    ],
+    function(declare,
+             Deferred,
+             _,
+             xmlToJson,
+             ajaxUtil,
+             util,
+             LayerNode) {
+        "use strict";
+
+        return declare(null, {
+            constructor: function(server) {
+                this.server = server;
+            },
+
+            getServiceUrl: function() {
+                return util.urljoin(this._getBaseServiceUrl() + '?request=GetCapabilities&service=WMS');
+            },
+
+            _getBaseServiceUrl: function() {
+                // Likely an AGS WMS service
+                if (this.server.name) {
+                    return util.urljoin(this.server.url, this.server.name, 'MapServer/WMSServer');
+                } else {
+                    return this.server.url;
+                }
+            },
+
+            // Return a promise containing map service data.
+            fetchMapService: function() {
+                return ajaxUtil.fetch(this.getServiceUrl(), 'text', this._transformServiceData);
+            },
+
+            // Return cached map service data.
+            getServiceData: function() {
+                return ajaxUtil.get(this.getServiceUrl());
+            },
+
+            // Return a promise with service layer data.
+            fetchLayerDetails: function(tree, layerId) {
+                return new Deferred().resolve(function() {
+                    return tree.findLayer(layerId);
+                });
+            },
+
+            // Return cached layer details.
+            getLayerDetails: function(serviceLayer) {
+                return new Deferred().resolve(function() {
+                    return tree.findLayer(layerId);
+                });
+            },
+
+            // Find the corresponding data for `layer` in the map service.
+            findServiceLayer: function(layer) {
+                var serviceData = this.getServiceData();
+
+                if (!serviceData || !layer) {
+                    return null;
+                }
+
+                return _.find(serviceData.layers, function(serviceLayer) {
+                    if (layer.getName() === serviceLayer.name) {
+                        // Compare not only the name, but the structure as well.
+                        // Protects against an edge case where a map service
+                        // contains a parent and child layer with the same name.
+                        if (layer.hasChildren() && serviceLayer.subLayerIds) {
+                            return true;
+                        } else if (!layer.hasChildren() && !serviceLayer.subLayerIds) {
+                            return true;
+                        }
+                    }
+                    return false;
```

Comment:
This issue might not exist in WMS so we may be able to simplify this comparison.